### PR TITLE
Add visualization plugin for strain rate tensor

### PR DESCRIPTION
**Pull request by bartniday**
Adds a visualization that outputs the strain rate tensor (called `strain rate tensor` because the existing `strain rate` outputs the norm of the strain rate tensor).

Errors in the documentation are likely.

### CODE REVIEW

File: source/postprocess/visualization/strain_rate_tensor.cc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        {}
+
+      template <int dim>
+      void
+      StrainRateTensor<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double> > &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points,    ExcInternalError());
```
Comment (gassmoeller) [inline source/postprocess/visualization/strain_rate_tensor.cc]: Please remove the extra spaces.

File: source/postprocess/visualization/strain_rate_tensor.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        Assert ((computed_quantities[0].size() == Tensor<2,dim>::n_independent_components),
+                ExcInternalError());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components,   ExcInternalError());
```
Comment (gassmoeller) [inline source/postprocess/visualization/strain_rate_tensor.cc]: You can remove this Assert as you do not access the `solution_values` anyway.

File: source/postprocess/visualization/strain_rate_tensor.cc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(StrainRateTensor,
+                                                  "strain rate tensor",
+                                                  "A visualization output object that generates output "
+                                                  "for the 4 (in 2d) or 9 (in 3d) components of the strain rate "
+                                                  "tensor, i.e., for the components of the tensor "
+                                                  "$\\varepsilon(\\mathbf u)+pI$ "
```
Comment (gassmoeller) [inline source/postprocess/visualization/strain_rate_tensor.cc]: Should it not be only `$\\varepsilon(\\mathbf u)$` (without the `+pI`)?

**Comment by gassmoeller** [comment]
Also please add an entry into doc/modules/changes about the new postprocessor, and add a picture and a few sentences into the hackathon report.

**Comment by tjhei** [comment]
Looks good!

**Comment by tjhei** [comment]
compilation failed:
```

In file included from /home/dealii/source/source/postprocess/visualization/strain_rate_tensor.cc:22:0:
/home/dealii/source/include/aspect/postprocess/visualization/strain_rate_tensor.h:52:41: error: expected template-name before â€˜<â€™ token
         : public DataPostprocessorTensor<dim>,
```

**Comment by bangerth** [comment]
I think `DataPostprocessorTensor` is a class that only entered deal.II in version 9.0. You may have to make the whole class you add here conditional on having deal.II 9.0.

File: include/aspect/postprocess/visualization/strain_rate_tensor.h [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_postprocess_visualization_strain_rate_tensor_h
+#define _aspect_postprocess_visualization_strain_rate_tensor_h
+
+#if DEAL_II_VERSION_GTE(9,0,0)
```
Comment (bangerth) [inline include/aspect/postprocess/visualization/strain_rate_tensor.h]: This doesn't compile (and same for the .cc file):
```
/home/dealii/source/include/aspect/postprocess/visualization/strain_rate_tensor.h:25:24: error: missing binary operator before token "("
 #if DEAL_II_VERSION_GTE(9,0,0)
                        ^
```
The reason is that it doesn't know about that macro. You'll have to move this `#if` statement *after* the `#include` files for this to work.

File: include/aspect/postprocess/visualization/strain_rate_tensor.h [hunk 5]
```diff
@@ -0,0 +1,69 @@
+/*
+  Copyright (C) 2015 - 2017 by the authors of the ASPECT code.
```
Comment (bangerth) [inline include/aspect/postprocess/visualization/strain_rate_tensor.h]: 2018

File: source/postprocess/visualization/strain_rate_tensor.cc [hunk 6]
```diff
@@ -0,0 +1,96 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
```
Comment (bangerth) [inline source/postprocess/visualization/strain_rate_tensor.cc]: 2018

**Comment by bartniday** [comment]
Compiler errors are fixed now

**Pull request merged**